### PR TITLE
ARROW-7279: [C++] Rename UnionArray::type_ids to type_codes

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

File: cpp/src/arrow/array.cc [hunk 1]
```diff
@@ -704,7 +704,7 @@ void UnionArray::SetData(const std::shared_ptr<ArrayData>& data) {
 
   auto type_ids = data_->buffers[1];
```
Comment (kou) [inline cpp/src/arrow/array.cc]: `type_ids` -> `type_codes`?

File: cpp/src/arrow/array/validate.cc [hunk 2]
```diff
@@ -334,15 +334,14 @@ struct ValidateArrayDataVisitor {
...
     const auto& child_ids = array.union_type()->child_ids();
 
-    const int8_t* type_ids = array.raw_type_ids();
+    const int8_t* type_codes = array.raw_type_codes();
     for (int64_t i = 0; i < array.length(); ++i) {
       if (array.IsNull(i)) {
         continue;
       }
-      const int32_t id = type_ids[i];
+      const int32_t id = type_codes[i];
```
Comment (kou) [inline cpp/src/arrow/array/validate.cc]: `id` -> `code`?

File: cpp/src/arrow/array/validate.cc [hunk 3]
```diff
@@ -362,7 +362,7 @@ struct ValidateArrayDataVisitor {
         if (array.IsNull(i)) {
           continue;
         }
-        const int32_t id = type_ids[i];
+        const int32_t id = type_codes[i];
```
Comment (kou) [inline cpp/src/arrow/array/validate.cc]: `id` -> `code`?

File: cpp/src/arrow/compute/kernels/take_internal.h [hunk 4]
```diff
@@ -603,7 +603,7 @@ class TakerImpl<IndexSequence, UnionType> : public Taker<IndexSequence> {
       // Fill child_offsets_storage with the taken offsets
       RETURN_NOT_OK(offset_builder_->Reserve(indices.length()));
       RETURN_NOT_OK(VisitIndices(indices, values, [&](int64_t index, bool is_valid) {
-        auto type_id = type_ids[index];
+        auto type_id = type_codes[index];
```
Comment (kou) [inline cpp/src/arrow/compute/kernels/take_internal.h]: `type_id` -> `type_code`?

File: cpp/src/arrow/pretty_print.cc [hunk 5]
```diff
@@ -354,7 +354,7 @@ class ArrayPrinter : public PrettyPrinter {
 
     Newline();
     Write("-- type_ids: ");
-    UInt8Array type_ids(array.length(), array.type_ids(), nullptr, 0, array.offset());
+    UInt8Array type_ids(array.length(), array.type_codes(), nullptr, 0, array.offset());
```
Comment (kou) [inline cpp/src/arrow/pretty_print.cc]: `type_ids` -> `type_codes`?

File: cpp/src/arrow/pretty_print.cc [hunk 6]
```diff
@@ -354,7 +354,7 @@ class ArrayPrinter : public PrettyPrinter {
 
     Newline();
     Write("-- type_ids: ");
```
Comment (kou) [inline cpp/src/arrow/pretty_print.cc]: `type_ids` -> `type_codes`?

File: cpp/src/arrow/python/deserialize.cc [hunk 7]
```diff
@@ -249,10 +249,10 @@ Status DeserializeSequence(PyObject* context, const Array& array, int64_t start_
       RETURN_NOT_OK(set_item(result.obj(), i - start_idx, Py_None));
     } else {
       int64_t offset = value_offsets[i];
-      uint8_t type = type_ids[i];
+      uint8_t type = type_codes[i];
       PyObject* value;
       RETURN_NOT_OK(GetValue(context, *data.child(type), offset,
-                             python_types[type_ids[i]], base, blobs, &value));
+                             python_types[type_codes[i]], base, blobs, &value));
```
Comment (kou) [inline cpp/src/arrow/python/deserialize.cc]: `type_codes[i]` -> `type`

File: ruby/red-arrow/ext/arrow/converters.hpp [hunk 8]
```diff
@@ -504,7 +504,7 @@ namespace red_arrow {
     uint8_t compute_child_index(const arrow::UnionArray& array,
                                 arrow::UnionType* type,
                                 const char* tag) {
-      const auto type_id = array.raw_type_ids()[index_];
+      const auto type_id = array.raw_type_codes()[index_];
```
Comment (kou) [inline ruby/red-arrow/ext/arrow/converters.hpp]: `type_id` -> `type_code`

**Review by kou — APPROVED** [review]
+1

**Comment by pitrou** [comment]
Thanks @kou . I think I addressed your comments now.

**Comment by pitrou** [comment]
AppVeyor: https://ci.appveyor.com/project/pitrou/arrow/builds/29306927

**Comment by wesm** [comment]
+1

**Pull request closed without merging**